### PR TITLE
canonicalize for InfiniteMPS

### PR DESCRIPTION
**Pull request by mganahl**
This PR depends on the following PRs:
* #382
* #385 
* #386

### CODE REVIEW

**Comment by Thenerdstation** [comment]
Fix the merge conflicts and I'll start a review.

File: tensornetwork/matrixproductstates/infinite_mps.py [hunk 1]
```diff
@@ -174,3 +175,147 @@ def left_envs(self, sites: Sequence[int]) -> Dict:
...
+      left_initial_state: Optional[Union[BaseNode, Tensor]] = None,
+      right_initial_state: Optional[Union[BaseNode, Tensor]] = None,
+      precision: Optional[float] = 1E-10,
+      truncation_threshold: Optional[float] = 1E-15,
+      D: Optional[int] = None,
+      num_krylov_vecs: Optional[int] = 50,
+      maxiter: Optional[int] = 1000,
+      pseudo_inverse_cutoff: Optional[float] = None):
+    """
+    Canonicalize an InfiniteMPS (i.e. bring it into Schmidt-canonical form).
```
Comment (amilsted) [inline tensornetwork/matrixproductstates/infinite_mps.py]: What is "Schmidt-canonical" form?

File: docs/mps.rst [hunk 2]
```diff
@@ -14,4 +14,7 @@ Matrix Product States
        
 .. autoclass:: tensornetwork.matrixproductstates.infinite_mps.InfiniteMPS
     :members:
+       
+.. autoclass:: tensornetwork.matrixproductstates.mps.InfiniteMPS
+    :members:
```
Comment (Thenerdstation) [inline docs/mps.rst]: Did you mean to read add this?

File: tensornetwork/matrixproductstates/infinite_mps_test.py [hunk 3]
```diff
@@ -79,3 +79,13 @@ def test_unitcell_transfer_operator(dtype, direction):
...
+
+
+@pytest.mark.parametrize("dtype", [np.float64, np.complex128])
+def test_InfiniteMPS_canonicalize(dtype):
+  D, d, N = 10, 2, 4
+  imps = InfiniteMPS.random(
+      d=[d] * N, D=[D] * (N + 1), dtype=dtype, backend='numpy')
+
+  imps.canonicalize()
+  assert imps.check_canonical() < 1E-12
```
Comment (Thenerdstation) [inline tensornetwork/matrixproductstates/infinite_mps_test.py]: Can you test this on more backends or is there something you are using that only works for numpy right now?

Comment (mganahl) [reply tensornetwork/matrixproductstates/infinite_mps_test.py]: this works only for numpy right now.

File: tensornetwork/matrixproductstates/infinite_mps.py (hunk 1 above)
Comment (mganahl) [reply tensornetwork/matrixproductstates/infinite_mps.py]: the canonical form

Comment (mganahl) [reply tensornetwork/matrixproductstates/infinite_mps.py]: Gamma Lambda ...

File: docs/mps.rst (hunk 2 above)
Comment (mganahl) [reply docs/mps.rst]: removed

File: tensornetwork/matrixproductstates/infinite_mps_test.py (hunk 3 above)
Comment (mganahl) [reply tensornetwork/matrixproductstates/infinite_mps_test.py]: opening an issue to address this

**Pull request merged**